### PR TITLE
Moving p2 deployment from Sourceforge to GitHub

### LINKED ISSUE #1204: Switching p2 deployment from SF to GH
so far p2 repositories are deployed to Sourceforge instead of GH.

The reason was, that sourceforge provides a simple ftp upload and https browsable access to these files.

Now with Tycho 4.0.8 our downstream builds e.g. CEF cannot suggest the p2 repositories anymore. even though resolved in the target file. The metadata caching of tycho breaks with a message "destFile" is not a file.

But it is possible to zip a p2 and place it into the target definition. This is digested by overtarget the eclipse p2 mechanisms as well as the tycho p2 mechanisms.

Therefore the p2 repositories should be attached as zipped artifacts to the releases on the GH platform.

### PR DESCRIPTION
**Pull request by ghost**
For more information see #1204 

Closes #1204

### CODE REVIEW

File: maven/pom.xml [hunk 1]
```diff
@@ -24,14 +24,11 @@
...
-		<build.folder.p2>p2</build.folder.p2>
-		<build.folder.bin>bin</build.folder.bin>
 		<build.folder.target.secured>../deploy/secured</build.folder.target.secured>
 		<build.folder.target.unsecured>../deploy/unsecured</build.folder.target.unsecured>
-		<build.folder.target.bin>../builds/${build.folder.bin}</build.folder.target.bin>
-		<build.folder.target.p2>../builds/${build.folder.p2}</build.folder.target.p2>
+		<build.folder.target>../builds/</build.folder.target>
 		<build.folder.type>test</build.folder.type>
-		<build.file.zip>VirSat_Core_Test.zip</build.file.zip>
+		<build.folder.fullpath>${build.folder.target}/${build.folder.type}</build.folder.fullpath>
```
Comment (franzTobiasDLR) [inline maven/pom.xml:31]: This looks like a placeholder value?

**Review by franzTobiasDLR — APPROVED** [review]
Nice, seems to reduce complexity of our build process quite a lot!

File: launchers/VirSat Core Build Development (Deploy Unsecured).launch [hunk 2]
```diff
@@ -14,6 +14,9 @@
     <stringAttribute key="M2_USER_SETTINGS" value=""/>
     <booleanAttribute key="M2_WORKSPACE_RESOLUTION" value="false"/>
     <booleanAttribute key="org.eclipse.debug.core.ATTR_FORCE_SYSTEM_CONSOLE_ENCODING" value="false"/>
+    <mapAttribute key="org.eclipse.debug.core.environmentVariables">
+        <mapEntry key="GITHUB_SHA" value="12345ABCDEF"/>
```
Comment (franzTobiasDLR) [inline launchers/VirSat Core Build Development (Deploy Unsecured).launch:18]: This looks like a placeholder value?

Comment (PhilMFischer) [reply launchers/VirSat Core Build Development (Deploy Unsecured).launch:18]: Yes, the build scripts (in maven pom) are using it for release builds. If oyu want to test run a release build locally, it is better to have it set.

**Review by kain-ap — APPROVED** [review]
Looks Great !

**Pull request merged**